src/test/FuzzerImpl.cpp

```diff
@@ -810,6 +810,46 @@ generator_t::operator()(stellar::PublicKey& t) const
         t.ed25519(), static_cast<uint8_t>(stellar::rand_uniform<int>(
                          0, NUMBER_OF_ACCOUNT_IDS_TO_GENERATE - 1)));
 }
+
+static int RECURSION_COUNT = 0;
+static const int RECURSION_LIMIT = 50;
+
+template <>
+void
+generator_t::operator()<stellar::SCVal>(stellar::SCVal& val) const
+{
+    if (++RECURSION_COUNT > RECURSION_LIMIT)
+    {
+        stellar::SCVal v;
+        val = v;
+        return;
+    }
+    const auto& vals = stellar::SCVal::_xdr_case_values();
+    stellar::SCValType v;
+
+    uint32_t n;
+    (*this)(n);
+    v = vals[n % vals.size()];
```

Comment:
Do we have fuzzing scenarios where it would make sense to generate arbitrary discriminators?

Reply:
Do you mean invalid discriminators? They wouldn't be valid xdr so no.

Reply:
Sure, so e.g. at the overlay layer or at tx submission layer that would be a valid scenario to fuzz (because we don't want to crash on invalid XDR). If we're skipping these layers, then only valid XDR would make sense indeed.

Reply:
Ah if we wanted to test invalid XDR for the overlay fuzzer, then that would need to be somewhere outside of xdrpp. The fuzzer should end up testing this though since it modifies the tests while it fuzzes.

Reply:
Sounds good, thanks for the confirmation.